src/transformers/models/wav2vec2/modeling_wav2vec2.py

```diff
@@ -67,50 +67,43 @@ def _compute_mask_indices(
     Adapted from `fairseq's data_utils.py
     <https://github.com/pytorch/fairseq/blob/e0788f7007a8473a76db573985031f3c94201e79/fairseq/data/data_utils.py#L376>`__.
     """
+    torch_device = "cuda" if torch.cuda.is_available() else "cpu"
     bsz, all_sz = shape
-    mask = np.full((bsz, all_sz), False)
-
-    all_num_mask = int(
-        # add a random number for probabilistic rounding
-        mask_prob * all_sz / float(mask_length)
-        + np.random.rand()
-    )
-
-    all_num_mask = max(min_masks, all_num_mask)
-
-    mask_idcs = []
+    mask = torch.full((bsz, all_sz), False, device=torch_device)
+    min_len_unique, mask_idcs = float("Inf"), []
     padding_mask = attention_mask.ne(1) if attention_mask is not None else None
+    if padding_mask is not None:
+        sz = all_sz - padding_mask.long().sum(dim=1)
+        num_mask = (((sz * mask_prob) / mask_length) + torch.rand(bsz, device=torch_device)).int()
+    else:
+        sz = torch.full([bsz], all_sz, device=torch_device)
+        all_num_mask = ((all_sz * mask_prob) / mask_length) + torch.rand(1, device=torch_device)[0]
+        num_mask = torch.full([bsz], all_num_mask, device=torch_device).int()
+
+    num_mask = torch.clamp(num_mask, min=min_masks)
```

Comment:
any reason to not use `torch.max` here?

Reply:
As we have vectorized the following process: `num_mask = max(min_masks, num_mask)` checked if a value was greater than a `min_mask` value, if it was we select that number otherwise, we select `min_masks` value, this process was used for a single integer value, but we have a Tensor so to do this operation on all the elements of the tensors, we have performed the [clamp](https://pytorch.org/docs/stable/generated/torch.clamp.html) operation. We could also do something like this: `num_mask[num_mask<min_masks] = min_masks`